### ISSUE TITLE
NPE when updating Discord activity

### ISSUE THREAD

**Issue by jungltemple** [OfficialDonut/VisualBukkit#112]
Visual Bukkit doesn't seem to be loading after it tries to invoke the discord thing. Here's my log:

```
Feb 19, 2024 8:20:35 AM com.gmail.visualbukkit.VisualBukkitApp start
INFO: Visual Bukkit v6.0.2
Feb 19, 2024 8:20:36 AM com.gmail.visualbukkit.VisualBukkitApp start
INFO: Loading extensions
Feb 19, 2024 8:20:36 AM com.gmail.visualbukkit.VisualBukkitApp start
INFO: Loading initial project
Feb 19, 2024 8:20:38 AM com.gmail.visualbukkit.VisualBukkitGrpcServer start
INFO: gRPC server running on port 50051 (unencrypted)
Feb 19, 2024 8:20:39 AM com.gmail.visualbukkit.VisualBukkitApp displayException
SEVERE: An unexpected error occurred
java.lang.NullPointerException: Cannot invoke "discord_game_sdk.IDiscordCore$get_activity_manager_callback.apply(discord_game_sdk.IDiscordCore)" because "core[0].get_activity_manager" is null
	at com.gmail.visualbukkit.VisualBukkitApp.updateDiscordActivity(VisualBukkitApp.java:456)
	at com.gmail.visualbukkit.VisualBukkitApp.lambda$start$26(VisualBukkitApp.java:265)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:185)
	at java.base/java.lang.Thread.run(Thread.java:1583)

```
And nothing else happens. I only have the green, logic-type blocks in the app. I don't know if this is the actual cause, but it's the only error I can find.

Here is a screenshot of what I see in the app:
![image](https://github.com/OfficialDonut/VisualBukkit/assets/146572102/4d76fc97-d12a-447a-9f13-b69e58325e32)

**Comment by Sal4iDev** [comment]
everything in the app that you see is fine. the problem with discordsdk, everything else is fine

**Comment by Sal4iDev** [comment]
everything in the app that you see is fine. the problem with discordsdk, everything else is fine

**Comment by jungltemple** [comment]
Ok, I'm confused. I don't have any of the blocks that are always shown in screenshots. Or maybe I just don't know how to get them...

**Comment by Sal4iDev** [comment]
> Ok, I'm confused. I don't have any of the blocks that are always shown in screenshots. Or maybe I just don't know how to get them...

it's vb6, not vb5 and lower. all the methods are in the Method block, fields in the Field block, etc

**Comment by jungltemple** [comment]
That makes more sense. Thank you.

**Comment by Sal4iDev** [comment]
leave it open, the problem with discord has not gone away, or..?

**Comment by jungltemple** [comment]
> leave it open, the problem with discord has not gone away, or..?

I don't care about discord.

**Comment by Sal4iDev** [comment]
yea but its vb bug, just leave the Issue open, Donut will fix it. It may be not only your problem

**Comment by jungltemple** [comment]
> yea but its vb bug, just leave the Issue open, Donut will fix it. It may be not only your problem

If you say so.

**Comment by OfficialDonut** [comment]
Fixed in 6.0.3
